crates/vm/src/arch/execution_mode/metered/mod.rs

```diff
@@ -255,13 +282,11 @@ where
         // Check if segmentation needs to happen
         self.check_segment_limits::<F, VC>(state, chip_complex);
 
-        let mut offset = if chip_complex.config().has_public_values_chip() {
-            PUBLIC_VALUES_AIR_ID + 1
+        let offset = if chip_complex.config().has_public_values_chip() {
+            PUBLIC_VALUES_AIR_ID + 1 + chip_complex.memory_controller().num_airs()
         } else {
-            PUBLIC_VALUES_AIR_ID
+            PUBLIC_VALUES_AIR_ID + chip_complex.memory_controller().num_airs()
         };
-        offset += chip_complex.memory_controller().num_airs();
```

Comment:
nit: not sure why you changed this, it seems harder to read

Reply:
i think i'll keep this as is. it doesn't need to be mutable